### PR TITLE
Style

### PR DESCRIPTION
**Pull request by levb**
- Moved things around, no functional changes
  - /api -> apps
  - /apps -> apps/impl
  - /store -> apps/store
- Small style fixes

### CODE REVIEW

File: server/http/helloapp/n_user_joined_channel.go [hunk 1]
```diff
@@ -3,12 +3,11 @@ package helloapp
 import (
 	"net/http"
 
-	"github.com/mattermost/mattermost-plugin-apps/server/api"
 	"github.com/mattermost/mattermost-plugin-apps/server/apps"
 )
 
-func (h *helloapp) handleUserJoinedChannel(w http.ResponseWriter, req *http.Request,
-	claims *apps.JWTClaims, n *api.Notification) (int, error) {
+func (h *helloapp) nUserJoinedChannel(w http.ResponseWriter, req *http.Request,
```
Comment (jfrerich) [inline server/http/helloapp/n_user_joined_channel.go:9]: It's pretty intuitive using "f_xx", but I'm not sure "n_" is.  just a note and that we might consider.  Non-blocking

**Review by jfrerich — COMMENTED** [review]
LGTM!

File: server/http/helloapp/n_user_joined_channel.go (hunk 1 above)
Comment (levb) [reply server/http/helloapp/n_user_joined_channel.go:9]: these are "functions" and "notifications", probably worth expanding a bit 👍

File: server/apps/api.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+type JWTClaims struct {
+	jwt.StandardClaims
+	ActingUserID string `json:"acting_user_id,omitempty"`
+}
+
+type InInstallApp struct {
+	GrantedPermissions Permissions
+	AppSecret          string
+	OAuth2TrustedApp   bool
```
Comment (mickmister) [inline server/apps/api.go:48]: Just an aside: If these are serialized, should we have json tags on the fields?

**Review by mickmister — APPROVED** [review]
LGTM :+1:

File: server/http/helloapp/n_user_joined_channel.go (hunk 1 above)
Comment (levb) [reply server/http/helloapp/n_user_joined_channel.go:9]: Added a comment

File: server/apps/api.go (hunk 2 above)
Comment (levb) [reply server/apps/api.go]: done

**Pull request merged**